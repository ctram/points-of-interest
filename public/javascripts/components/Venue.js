import React from "react";

import "../../stylesheets/Venue.css";

function Address(props) {
  const {
    location: { address, city, postalCode, state }
  } = props;

  let cityStatePostalCode = "";

  cityStatePostalCode += city || "";

  if (state) {
    cityStatePostalCode += ", " + state;
  }

  if (postalCode) {
    cityStatePostalCode += " " + postalCode;
  }

  return (
    <div className="address card-text">
      <div>{address}</div>
      <div>{cityStatePostalCode}</div>
      <div></div>
    </div>
  );
}

export default function Venue(props) {
  const {
    venue: { name, categories, location }
  } = props;

  let categoryTags = [];
  let domAddress = null;

  if (categories && categories.length) {
    categoryTags = categories.map((category, idx) => {
      return (
        <span
          className="badge badge-secondary venue__tag"
          key={category.id || idx}
        >
          {category.name}
        </span>
      );
    });
  }

  if (location) {
    domAddress = <Address location={location} />;
  }

  return (
    <div className="venue card">
      <div className="card-body">
        <h5 className="card-title">{name}</h5>

        {categoryTags.length !== 0 && (
          <h6 className="card-subtitle mb-2 text-muted venue__tags">
            {categoryTags}
          </h6>
        )}

        {domAddress}
      </div>
    </div>
  );
}
