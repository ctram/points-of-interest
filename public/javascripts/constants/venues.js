const SET_VENUES = "SET_VENUES";

export { SET_VENUES };
