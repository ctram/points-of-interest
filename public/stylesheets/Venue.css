.venue {
  width: 50%;
  margin-bottom: 30px;
}

.venue .venue__tags {
  margin: 10px 0;
}

.venue .venue__tag {
  margin: 0 10px 0 0;
}
