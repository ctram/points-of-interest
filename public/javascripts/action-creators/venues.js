import { SET_VENUES } from "../constants/venues";

import FourSquareHelper from "../lib/four-square-helper";

const fourSquareClient = new FourSquareHelper();

export function fetchVenues({ coordinates, near, query, limit }) {
  return dispatch => {
    return fourSquareClient
      .nearByVenues({ coordinates, near, query, limit })
      .then(venues => {
        return dispatch({ type: SET_VENUES, venues });
      })
      .catch(e => {
        console.error(e);
      });
  };
}
