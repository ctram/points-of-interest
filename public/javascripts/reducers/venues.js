import { SET_VENUES } from "../constants/venues";

const initialState = [];

export default function venues(state = initialState, action) {
  const { type, venues } = action;

  switch (type) {
    case SET_VENUES:
      return venues;
    default:
      return state;
  }
}
