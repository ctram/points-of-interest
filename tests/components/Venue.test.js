import React from "react";
import Venue from "../../public/javascripts/components/Venue";
import renderer from "react-test-renderer";

import venues from "../mock-data/venues";

test("Venue component shows details", () => {
  const component = renderer.create(<Venue venue={venues[0]} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
