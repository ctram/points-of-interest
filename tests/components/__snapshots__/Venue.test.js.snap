// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue component shows details 1`] = `
<div
  className="venue card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Hot Shot
    </h5>
    <h6
      className="card-subtitle mb-2 text-muted venue__tags"
    >
      <span
        className="badge badge-secondary venue__tag"
      >
        Cafe
      </span>
      <span
        className="badge badge-secondary venue__tag"
      >
        Bar
      </span>
    </h6>
    <div
      className="address card-text"
    >
      <div>
        123 Somewhere
      </div>
      <div>
        Gotham, NY 12345
      </div>
      <div />
    </div>
  </div>
</div>
`;
